### ISSUE TITLE
Ability to return errors with extra fields

### ISSUE THREAD

**Issue by F21** [neelance/graphql-go#37]
According to the [spec](http://facebook.github.io/graphql/#sec-Errors), errors must contain a `message` key. However, the developer can add other fields if he wants.

In my case, for each error, I want to return both a `message` which is developer friendly, and an `error_code` such as `USERNAME_UNAVAILABLE`. It would be nice if returning errors with extra user defined fields is possible.

**Comment by neelance** [comment]
This got resolved by #49, especially see https://github.com/neelance/graphql-go/pull/49#issuecomment-271729496.

**Comment by F21** [comment]
Hey @neelance 

I am a bit confused on how to use the `QueryError`. In my case, I want to return errors with  message and code, for example: 

```json
{
   ...
  errors: [
    {"key": "some_error", "msg": "some message"},
    ...
  ]
}
```

Is there an example for using `QueryError`?
